### PR TITLE
Preprocess theme CSS to reduce number of files

### LINKED ISSUE #3798: Icons loading on demand slows their rendering
I confirmed what @ian-r-rose observed this morning that our SVG icons are being loaded from the server on demand. When working on remote JupyterLab instances, this can take a long enough that various icons being to be glitchy.

Is there a way to ship these inline?

### PR DESCRIPTION
**Pull request by saulshanabrook**
Fixes #3798 

Preprocesses each CSS theme in webpack, to optimize it's CSS bundle. This combines all the CSS files and lines the SVG and PNG files. It also optimizes the SVG files, which I found reduces the final `index.css` (in the light theme) from 196 kB to 67 kB.

### CODE REVIEW

**Comment by blink1073** [comment]
I tested this locally, and it works great, thanks!  We need a check whether to build the theme [here](https://github.com/jupyterlab/jupyterlab/blob/466969c3114e5a89ab15796ed3582a9a6b70e4b4/jupyterlab/commands.py#L92) and [here](https://github.com/jupyterlab/jupyterlab/blob/466969c3114e5a89ab15796ed3582a9a6b70e4b4/jupyterlab/commands.py#L107).

**Comment by saulshanabrook** [comment]
@blink1073 I was able to add it to `ensure_dev` and checked that it worked during `jupyter lab --dev-mode --watch`. However, I am not sure when `ensure_core` is run. I tried running `jupyter lab --watch` and `jupyter lab` but neither of broke when the theme static files were gone and didn't use my local `commands.py` file. Is that because they are running jupyter lab globally? Can I run it using my local repo in core mode?

**Comment by blink1073** [comment]
Hi @saulshanabrook, you can run core mode by running `jupyter lab --core-mode`.  The other commands are only using released JavaScript assets, so wouldn't include your changes to the them files.

**Comment by saulshanabrook** [comment]
@blink1073 Ah ok thank you. Now I can't seem to get it to load my local theme changes when running in `--core-mode`. I have tried doing `yarn run clean; yarn run clean:slate; pip3 install -e .` but it still seems to be loading the old theme files.

**Comment by blink1073** [comment]
Right, core mode also uses released assets, my mistake.

File: jupyterlab/commands.py [hunk 1]
```diff
@@ -98,6 +98,14 @@ def ensure_dev(logger=None):
         yarn_proc = Process(['node', YARN_PATH], cwd=parent, logger=logger)
         yarn_proc.wait()
 
+    theme_packages = ['theme-light-extension', 'theme-dark-extension']
```
Comment (blink1073) [inline jupyterlab/commands.py]: Can we use `build:themes` here from the top level?

Comment (saulshanabrook) [reply jupyterlab/commands.py]: We can, but then they will be built unconditionally, even if they were already built. The other checks in this file only build if the files didn't already exist, so I tried to mirror those.

Comment (blink1073) [reply jupyterlab/commands.py]: Ah, fair enough.

**Comment by blink1073** [comment]
This LGTM, I tested locally.  Very nice!

File: jupyterlab/commands.py (hunk 1 above)
Comment (saulshanabrook) [reply jupyterlab/commands.py]: I was considering globbing for all theme packages using [pathlib](https://docs.python.org/3/library/pathlib.html#basic-use), but I was hesitant to differ from the os.path style of the rest of the file.

**Comment by saulshanabrook** [comment]
@blink1073 Thanks for the reviews!

**Pull request merged**